src/rust/engine/BUILD

```diff
@@ -20,16 +21,19 @@ _RELTYPE_FLAG = "" if env("MODE") == "debug" else "--release"
 shell_command(
     name="engine-and-client",
     command=(
-        'PATH="${PATH}:${CHROOT}/3rdparty/tools/python3/python/bin:${CHROOT}/3rdparty/tools/protoc/protoc/bin"'
+        'PATH="${PATH}:${CHROOT}/3rdparty/tools/python3/python/bin";'
+        + 'PATH="${PATH}:${CHROOT}/3rdparty/tools/protoc/protoc/bin";'
+        + 'PATH="${PATH}:${CHROOT}/3rdparty/tools/rust/cargo/bin";'
         + f" cargo build {_RELTYPE_FLAG} --features=extension-module -p engine -p client"
     ),
     execution_dependencies=[
         ":rust_sources",
         "3rdparty/tools/protoc:protoc",
         "3rdparty/tools/python3:python3",
+        "3rdparty/tools/rust:cargo",
     ],
     extra_env_vars=["CHROOT={chroot}"],
-    tools=["bash", "cc", "ld", "as", "ar", "cargo"],
```

Comment:
@huonw I know you have concerns. Let's work through them here. I'm happy to try almost anything.